order/views.py

```diff
@@ -1,3 +1,70 @@
-from django.shortcuts import render
+from django.contrib import messages
+from django.shortcuts import redirect
+from django.urls import reverse_lazy
+from django.views.generic import CreateView, UpdateView
+
+from cart.models import Cart
+from order.forms import CheckoutForm, CardPaymentForm
+from order.models import Order
+from user.models import User
+
+
+class CheckoutView(CreateView):
+    model = Order
+    template_name = 'order/checkout.html'
+    form_class = CheckoutForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['slug'] = self.request.user.slug
+        context['cost'] = self.request.user.cart.cost
+        return context
+
+    def get_initial(self):
+        initial = {
+            'name': self.request.user.name,
+            'address': self.request.user.address,
+            'phone': self.request.user.phone,
+            'user': self.request.user,
+            'products': self.request.user.cart.products,
+            'cost': self.request.user.cart.cost
+        }
+        return initial
+
+    def get_success_url(self):
+        return reverse_lazy('user:profile', args=[self.request.user.slug])
+
+    def form_valid(self, form):
```

Comment:
Better use post + some additional method to process this logic. Cause here any actions with form detected